internal/auth/issuer/pam/pam_issuer.go

```diff
@@ -0,0 +1,951 @@
+
+	// Check if user is already authenticated via session
+	if sessionID == "" {
+		s.log.Infof("Authorize: no session ID found, returning login form")
+		// User not authenticated, return embedded login form
+		loginForm := s.GetLoginForm(req.ClientId, req.RedirectUri, lo.FromPtrOr(req.State, ""))
+		return &AuthorizeResponse{
+			Type:    AuthorizeResponseTypeHTML,
+			Content: loginForm,
+		}, nil
+	}
+
+	// Check if session exists and is valid
+	sessionData, exists := s.IsUserAuthenticated(sessionID)
+	if !exists {
+		s.log.Infof("Authorize: session not found or expired - %s, returning login form", sessionID)
+		// Session invalid or expired, return login form
+		loginForm := s.GetLoginForm(req.ClientId, req.RedirectUri, lo.FromPtrOr(req.State, ""))
+		return &AuthorizeResponse{
+			Type:    AuthorizeResponseTypeHTML,
+			Content: loginForm,
+		}, nil
+	}
+
+	// User is authenticated, get username from session
+	username := sessionData.Username
+	s.log.Infof("Authorize: user authenticated via session - username=%s, sessionID=%s", username, sessionID)
+
+	// Generate authorization code
+	authCode, err := generateAuthorizationCode()
+	if err != nil {
+		s.log.Errorf("Authorize: failed to generate authorization code - %v", err)
+		return nil, fmt.Errorf("server_error")
+	}
+	s.log.Infof("Authorize: generated authorization code - %s", authCode)
+
+	// Use the requested scope if provided, otherwise determine based on user's role/group membership
+	scopes := lo.FromPtrOr(req.Scope, s.determineUserScopes(username))
+	s.log.Infof("Authorize: determined scopes - %s", scopes)
+
+	// Store authorization code with expiration (10 minutes)
+	codeData := &AuthorizationCodeData{
+		Code:        authCode,
+		ClientID:    req.ClientId,
+		RedirectURI: req.RedirectUri,
+		Scope:       scopes,
+		State:       lo.FromPtrOr(req.State, ""),
+		Username:    username,
+		ExpiresAt:   time.Now().Add(10 * time.Minute),
+		CreatedAt:   time.Now(),
+	}
+
+	s.codeStore.StoreCode(codeData)
+	s.log.Infof("Authorize: stored authorization code for user - %s", username)
+
+	// Build redirect URL with authorization code
+	parsed, err := url.Parse(req.RedirectUri)
+	if err != nil {
+		return nil, fmt.Errorf("invalid redirect URI: %w", err)
+	}
+
+	values := parsed.Query()
+	values.Set("code", authCode)
+	if req.State != nil && *req.State != "" {
+		values.Set("state", *req.State)
+	}
+	parsed.RawQuery = values.Encode()
+	redirectURL := parsed.String()
+
+	s.log.Infof("Authorize: returning redirect URL - %s", redirectURL)
+	return &AuthorizeResponse{
+		Type:    AuthorizeResponseTypeRedirect,
+		Content: redirectURL,
+	}, nil
+}
+
+// Login handles the login form submission
+func (s *PAMOIDCProvider) Login(ctx context.Context, username, password, clientID, redirectURI, state string) (*LoginResult, error) {
+	s.log.Infof("Login: attempting authentication for user %s, clientID=%s, redirectURI=%s", username, clientID, redirectURI)
```

Comment:
can we review all info level logs in this file if the information is appropriate